Development Basic Setup on Raspberry Pi OS (Buster)

Good job on the instructions!

To confirm, my distribution should be up to date.
```
sudo apt update
Hit:1 http://archive.raspberrypi.org/debian buster InRelease
Get:2 http://raspbian.raspberrypi.org/raspbian buster InRelease [15.0 kB]
Fetched 15.0 kB in 1s (15.9 kB/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done
All packages are up to date.
```

I've encountered two issues so far.

## Install Dependencies
Running
```
sudo apt install -y
```
it complained that `gcc-multilib` wasn't available which I bypassed by removing it from the command.

## Install Zephyr Python Dependencies

Running
```
pip3 install --user -r zephyr/scripts/requirements.txt
```
it complained

```
Collecting cmsis-pack-manager>=0.2.7 (from pyocd>=0.24.0->-r zephyr/scripts/requirements-run-test.txt (line 9))
  Downloading https://files.pythonhosted.org/packages/c9/2b/1ac9b85b5ba535ba386467cb9a897e3f460c04f34a305e02b22f86b65367/cmsis-pack-manager-0.2.10.tar.gz (45kB)
    100% |████████████████████████████████| 51kB 2.4MB/s
    Complete output from command python setup.py egg_info:
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    c/_cffi_backend.c:15:10: fatal error: ffi.h: No such file or directory
     #include <ffi.h>
              ^~~~~~~
    compilation terminated.
 ...
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-e44et2su/cmsis-pack-manager/
```

so I installed `libffi-dev`
```
sudo apt-get install libffi-dev
```
and then reran
```
pip3 install --user -r zephyr/scripts/requirements.txt
```
it then complained
```
  Downloading https://files.pythonhosted.org/packages/ae/e7/d9c3a176ca4b02024debf82342dab36efadfc5776f9c8db077e8f6e71821/pycparser-2.20-py2.py3-none-any.whl (112kB)
    100% |████████████████████████████████| 112kB 1.9MB/s
Building wheels for collected packages: cmsis-pack-manager
  Running setup.py bdist_wheel for cmsis-pack-manager ... error
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-45ylt83v/cmsis-pack-manager/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-bglvsvky --python-tag cp37:
  running bdist_wheel
  running build
  running build_py
  creating build/lib
  creating build/lib/cmsis_pack_manager
  copying cmsis_pack_manager/__init__.py -> build/lib/cmsis_pack_manager
  copying cmsis_pack_manager/pack_manager.py -> build/lib/cmsis_pack_manager
  copying cmsis_pack_manager/_native__ffi.py -> build/lib/cmsis_pack_manager
  copying cmsis_pack_manager/_version.py -> build/lib/cmsis_pack_manager
  copying cmsis_pack_manager/_native.py -> build/lib/cmsis_pack_manager
  error: [Errno 2] No such file or directory: 'cargo': 'cargo'

  ----------------------------------------
  Failed building wheel for cmsis-pack-manager
```
which I *thought* was due to an older version of pip, according to https://github.com/ARMmbed/cmsis-pack-manager/issues/137
So I upgraded pip:
```
python3 -m pip install --upgrade pip
```
but it didn't help the cmsis/cargo error.

I'm still looking into this.  Any thoughts please?

--- a/docs/docs/dev-setup.md
+++ b/docs/docs/dev-setup.md
@@ -36,20 +36,16 @@ A unix-like environment with the following base packages installed:
 <TabItem value="debian">
 On Debian and Ubuntu, we'll use apt to install our base dependencies:
 
-#### Apt Update
-
 First, if you haven't updated recently, or if this is a new install,
 you should update to get the latest package information:
 
-```bash
+```sh
 sudo apt update
 ```
 
-#### Install Dependencies
-
 With the latest package information, you can now install the base dependencies:
 
-```bash
+```sh
 sudo apt install -y \
 	git \
 	wget \
@@ -79,20 +75,16 @@ or download and install CMake version 3.13.1 or newer manually.
 <TabItem value="raspberryos">
 On Raspberry OS, we'll use apt to install our base dependencies:
 
-#### Apt Update
-
 First, if you haven't updated recently, or if this is a new install,
 you should update to get the latest package information:
 
-```bash
+```sh
 sudo apt update
 ```
 
-#### Install Dependencies
-
 With the latest package information, you can now install the base dependencies:
 
-```bash
+```sh
 sudo apt install -y \
 	git \
 	wget \
@@ -124,7 +116,7 @@ On Fedora, we'll use `dnf` to install our base dependencies:
 First, if you haven't updated recently, or if this is a new install,
 you should update to get the latest package information:
 
-```bash
+```sh
 sudo dnf update
 ```
 
@@ -132,7 +124,7 @@ sudo dnf update
 
 With the latest package information, you can now install the base dependencies:
 
-```bash
+```sh
 sudo dnf install -y \
 	git \
 	wget \
@@ -192,7 +184,7 @@ brew install cmake ninja python3 ccache dtc git wget
 
 West can be installed by using the `pip` python package manager.
 
-```bash
+```sh
 pip3 install --user -U west
 ```
 
@@ -235,13 +227,13 @@ the operating system's installed [cross compilers](https://docs.zephyrproject.or
 
 First, the cross compiler should be installed:
 
-```bash
+```sh
 sudo apt install gcc-arm-none-eabi
 ```
 
 Next, we'll configure Zephyr™ with some extra environment variables needed to find the cross compiler by adding the following to `~/.zephyrrc`:
 
-```bash
+```sh
 export ZEPHYR_TOOLCHAIN_VARIANT=cross-compile
 export CROSS_COMPILE=/usr/bin/arm-none-eabi-
 ```
@@ -304,13 +296,13 @@ Zephyr™ application is in the `app/` source directory:
 
 #### Step into the repository
 
-```bash
+```sh
 cd zmk
 ```
 
 #### Initialize West
 
-```bash
+```sh
 west init -l app/
 ```
 
@@ -322,19 +314,19 @@ section again for links to how to do this
 
 #### Update To Fetch Modules
 
-```bash
+```sh
 west update
 ```
 
 #### Export Zephyr™ Core
 
-```bash
+```sh
 west zephyr-export
 ```
 
 #### Install Zephyr Python Dependencies
 
-```bash
+```sh
 pip3 install --user -r zephyr/scripts/requirements-base.txt
 ```
 
@@ -432,7 +424,7 @@ cat ~/.zephyrrc >> ~/.zshrc
 
 From here on, building and flashing ZMK should all be done from the `app/` subdirectory of the ZMK checkout:
 
-```bash
+```sh
 cd app
 ```
 
@@ -451,7 +443,7 @@ Given the following:
 
 You can build ZMK with the following:
 
-```bash
+```sh
 west build -b proton_c -- -DSHIELD=kyria -DKEYMAP=default
 ```
 
@@ -466,7 +458,7 @@ Given the following:
 
 you can build ZMK with the following:
 
-```bash
+```sh
 west build -b planck -- -DKEYMAP=default
 ```
 

--- a/docs/docs/dev-setup.md
+++ b/docs/docs/dev-setup.md
@@ -9,7 +9,7 @@ import TabItem from '@theme/TabItem';
 
 export const OsTabs = (props) => (<Tabs
 groupId="operating-systems"
-defaultValue="linux"
+defaultValue="debian"
 values={[
 {label: 'Debian/Ubuntu', value: 'debian'},
 {label: 'Raspberry OS', value: 'raspberryos'},

--- a/docs/docs/dev-setup.md
+++ b/docs/docs/dev-setup.md
@@ -263,10 +263,27 @@ wget -q "https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v${ZSDK_
 The installation will prompt with several questions about installation location, and creating a default `~/.zephyrrc` for you with various variables. The defaults shouldn normally work as expected.
 
 </TabItem>
-<TabItem value="win">Windows instructions are coming soon!</TabItem>
+<TabItem value="win">
+
+#### GNU ARM Embedded
+
+Since the Zephyr™ SDK is not available for Windows, we recommending following the steps to install the [GNU ARM Embedded](https://docs.zephyrproject.org/latest/getting_started/toolchain_3rd_party_x_compilers.html#gnu-arm-embedded).
+
+</TabItem>
 <TabItem value="mac">
 
-Instructions for macOS coming soon.
+#### Zephyr™ ARM SDK
+
+To build firmwares for the ARM architecture (all supported MCUs/keyboards at this point), you'll need to install the Zephyr™ ARM SDK to your system:
+
+```
+export ZSDK_VERSION=0.11.2
+wget -q "https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v${ZSDK_VERSION}/zephyr-toolchain-arm-${ZSDK_VERSION}-setup.run" && \
+ sh "zephyr-toolchain-arm-${ZSDK_VERSION}-setup.run" --quiet -- -d ~/.local/zephyr-sdk-${ZSDK_VERSION} && \
+ rm "zephyr-toolchain-arm-\${ZSDK_VERSION}-setup.run"
+```
+
+The installation will prompt with several questions about installation location, and creating a default `~/.zephyrrc` for you with various variables. The defaults shouldn normally work as expected.
 
 </TabItem>
 </OsTabs>
@@ -330,26 +347,86 @@ We suggest two main [options](https://docs.zephyrproject.org/latest/guides/env_v
 
 To load the Zephyr environment properly for just one transient shell, run the following from your ZMK checkout directory:
 
+<OsTabs>
+<TabItem value="debian">
+
 ```
 source zephyr/zephyr-env.sh
 ```
+
+</TabItem>
+
+<TabItem value="raspberryos">
+
+```
+source zephyr/zephyr-env.sh
+```
+
+</TabItem>
+
+<TabItem value="fedora">
+
+```
+source zephyr/zephyr-env.sh
+```
+
+</TabItem>
+
+<TabItem value="mac">
+
+```
+source zephyr/zephyr-env.sh
+```
+
+</TabItem>
+
+<TabItem value="win">
+
+```
+source zephyr/zephyr-env.cmd
+```
+
+</TabItem>
+</OsTabs>
 
 #### All Shells
 
 To load the environment variables for your shell every time,
 append the existing `~/.zephyrrc` file to your shell's RC file and then start a new shell.
 
-##### Bash
+<Tabs
+groupId="shell"
+defaultValue="bash"
+values={[
+{label: 'bash', value: 'bash'},
+{label: 'zsh', value: 'zsh'},
+{label: 'cmd.exe', value: 'cmd'},
+]
+}>
+
+<TabItem value="bash">
 
 ```
 cat ~/.zephyrrc >> ~/.bashrc
 ```
 
-##### ZSH
+</TabItem>
+
+<TabItem value="zsh">
 
 ```
 cat ~/.zephyrrc >> ~/.zshrc
 ```
+
+</TabItem>
+
+<TabItem value="cmd">
+
+`cmd.exe` instructions coming soon!
+
+</TabItem>
+
+</Tabs>
 
 ## Build
 

--- a/docs/docs/dev-setup.md
+++ b/docs/docs/dev-setup.md
@@ -4,6 +4,21 @@ title: Basic Setup
 sidebar_label: Basic Setup
 ---
 
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+export const OsTabs = (props) => (<Tabs
+groupId="operating-systems"
+defaultValue="linux"
+values={[
+{label: 'Debian/Ubuntu', value: 'debian'},
+{label: 'Raspberry OS', value: 'raspberryos'},
+{label: 'Fedora', value: 'fedora'},
+{label: 'Windows', value: 'win'},
+{label: 'macOS', value: 'mac'},
+]
+}>{props.children}</Tabs>);
+
 ## Prerequisites
 
 A unix-like environment with the following base packages installed:
@@ -17,13 +32,13 @@ A unix-like environment with the following base packages installed:
 - `dfu-util`
 - Various build essentials, e.g. gcc, automake, autoconf
 
-### Debian/Ubuntu
-
+<OsTabs>
+<TabItem value="debian">
 On Debian and Ubuntu, we'll use apt to install our base dependencies:
 
 #### Apt Update
 
-First, if you haven't updated recently, or if this is a new install from WSL,
+First, if you haven't updated recently, or if this is a new install,
 you should update to get the latest package information:
 
 ```bash
@@ -60,41 +75,146 @@ sudo apt install -y \
 Ubuntu 18.04 LTS release packages a version of CMake that is too old. Please upgrade to Ubuntu 20.04 LTS
 or download and install CMake version 3.13.1 or newer manually.
 :::
+</TabItem>
+<TabItem value="raspberryos">
+On Raspberry OS, we'll use apt to install our base dependencies:
 
-### Fedora
+#### Apt Update
 
-TODO
+First, if you haven't updated recently, or if this is a new install,
+you should update to get the latest package information:
 
-### macOS
+```bash
+sudo apt update
+```
 
-TODO
+#### Install Dependencies
 
-### WSL
+With the latest package information, you can now install the base dependencies:
 
-Windows Subsystem for Linux can use various Linux distributions. Find a WSL installation on the [Windows Store](https://aka.ms/wslstore).
+```bash
+sudo apt install -y \
+	git \
+	wget \
+	autoconf \
+	automake \
+	build-essential \
+	ccache \
+	device-tree-compiler \
+	dfu-util \
+	g++ \
+	gcc \
+	libtool \
+	make \
+	ninja-build \
+	cmake \
+	python3-dev \
+	python3-pip \
+	python3-setuptools \
+	xz-utils
+```
 
-After installing your preferred flavor, follow the directions above on [Debian/Ubuntu](#debianubuntu) or [Fedora](#fedora).
+</TabItem>
+<TabItem value="fedora">
+
+On Fedora, we'll use `dnf` to install our base dependencies:
+
+#### DNF Update
+
+First, if you haven't updated recently, or if this is a new install,
+you should update to get the latest package information:
+
+```bash
+sudo dnf update
+```
+
+#### Install Dependencies
+
+With the latest package information, you can now install the base dependencies:
+
+```bash
+sudo dnf install -y \
+	git \
+	wget \
+	autoconf \
+	automake \
+	ccache \
+	dtc \
+	dfu-util \
+	g++ \
+	gcc \
+	libtool \
+	make \
+	ninja-build \
+	cmake \
+	python3-devel \
+	python3-pip \
+	python3-setuptools \
+	xz
+```
+
+</TabItem>
+<TabItem value="win">
 
 :::note
-On WSL2 don't put the project files into `/mnt/c/` as file I/O speeds are extremely slow. Instead, run everything in the Linux system and use `cp` to move files over to `/mnt/c/` as needed.
+Use `cmd.exe` with these instructions rather than PowerShell.
 :::
+
+Chocolatey is recommended and used for the following instructions. You can manually install each of these applications and add them to your `PATH` if you don't want to use Chocolatey.
+
+1. [Install Chocolatey](https://chocolatey.org/install)
+2. Open `cmd.exe` as **Administrator**
+3. Run the following `choco` commands:
+   ```shell
+   choco install cmake --installargs 'ADD_CMAKE_TO_PATH=System'
+   choco install ninja gperf python git
+   ```
+
+</TabItem>
+<TabItem value="mac">
+
+#### Homebrew
+
+Homebrew is required to install the system dependencies. If you haven't done so, visit [Homebrew](https://brew.sh/) for instructions. Once installed, use it to install the base dependencies:
+
+```
+brew install cmake ninja python3 ccache dtc git wget
+```
+
+</TabItem>
+</OsTabs>
 
 ## Setup
 
 ### West Build Command
 
-`west` is the [Zephyr™ meta-tool](https://docs.zephyrproject.org/latest/guides/west/index.html) used to configure and build Zephyr™ applications. It can be installed by using the `pip` python package manager:
+`west` is the [Zephyr™ meta-tool](https://docs.zephyrproject.org/latest/guides/west/index.html) used to configure and build Zephyr™ applications.
+
+West can be installed by using the `pip` python package manager.
 
 ```bash
-pip3 install --user west
+pip3 install --user -U west
 ```
 
-:::note
-If you don't already have it configured, you may need to update your
-`PATH` to include the pip install path. See [User Installs](https://pip.pypa.io/en/stable/user_guide/#user-installs) and [Stack Overflow](https://stackoverflow.com/questions/38112756/how-do-i-access-packages-installed-by-pip-user) for more details.
+:::tip pip user packages
+If you haven't done so yet, you may need to add the Python Pip user package directory to your `PATH`, e.g.:
+
+```
+echo 'export PATH=~/.local/bin:"$PATH"' >> ~/.bashrc
+source ~/.bashrc
+```
+
 :::
 
-### Zephyr™ ARM SDK
+### Toolchain Installation
+
+The toolchain provides the compiler, linker, etc necessary to build for the target
+platform.
+
+<OsTabs>
+<TabItem value="debian">
+
+#### Zephyr™ ARM SDK
 
 To build firmwares for the ARM architecture (all supported MCUs/keyboards at this point), you'll need to install the Zephyr™ ARM SDK to your system:
 
@@ -106,6 +226,50 @@ wget -q "https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v${ZSDK_
 ```
 
 The installation will prompt with several questions about installation location, and creating a default `~/.zephyrrc` for you with various variables. The defaults shouldn normally work as expected.
+
+</TabItem>
+<TabItem value="raspberryos">
+
+Because Raspberry OS (Raspbian) runs on the same architecture (but different ABI) as the keyboard MCUs,
+the operating system's installed [cross compilers](https://docs.zephyrproject.org/latest/getting_started/toolchain_other_x_compilers.html) can be used to target the different ABI.
+
+First, the cross compiler should be installed:
+
+```bash
+sudo apt install gcc-arm-none-eabi
+```
+
+Next, we'll configure Zephyr™ with some extra environment variables needed to find the cross compiler by adding the following to `~/.zephyrrc`:
+
+```bash
+export ZEPHYR_TOOLCHAIN_VARIANT=cross-compile
+export CROSS_COMPILE=/usr/bin/arm-none-eabi-
+```
+
+</TabItem>
+<TabItem value="fedora">
+
+#### Zephyr™ ARM SDK
+
+To build firmwares for the ARM architecture (all supported MCUs/keyboards at this point), you'll need to install the Zephyr™ ARM SDK to your system:
+
+```
+export ZSDK_VERSION=0.11.2
+wget -q "https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v${ZSDK_VERSION}/zephyr-toolchain-arm-${ZSDK_VERSION}-setup.run" && \
+ sh "zephyr-toolchain-arm-${ZSDK_VERSION}-setup.run" --quiet -- -d ~/.local/zephyr-sdk-${ZSDK_VERSION} && \
+ rm "zephyr-toolchain-arm-\${ZSDK_VERSION}-setup.run"
+```
+
+The installation will prompt with several questions about installation location, and creating a default `~/.zephyrrc` for you with various variables. The defaults shouldn normally work as expected.
+
+</TabItem>
+<TabItem value="win">Windows instructions are coming soon!</TabItem>
+<TabItem value="mac">
+
+Instructions for macOS coming soon.
+
+</TabItem>
+</OsTabs>
 
 ### Source Code
 

